datapipelines/src/test/java/io/customer/datapipelines/sdk/CustomerIOBuilderTest.kt

```diff
@@ -0,0 +1,135 @@
+package io.customer.datapipelines.sdk
+
+import android.app.Activity
+import android.app.Application
+import io.customer.commontest.BaseUnitTest
+import io.customer.commontest.module.CustomerIOGenericModule
+import io.customer.datapipelines.DataPipelinesModule
+import io.customer.datapipelines.extensions.apiHost
+import io.customer.datapipelines.extensions.cdnHost
+import io.customer.sdk.CustomerIOBuilder
+import io.customer.sdk.core.di.SDKComponent
+import io.customer.sdk.core.util.CioLogLevel
+import io.customer.sdk.data.model.Region
+import io.customer.sdk.extensions.random
+import org.amshove.kluent.shouldBe
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.never
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+import org.robolectric.Robolectric
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+class CustomerIOBuilderTest : BaseUnitTest() {
```

Comment:
I think there are a few tests missing:

- Validating default values
- Validating setting regions (both `US` and `EU`) works correctly and as expected